### PR TITLE
Master clean settings yti

### PR DESCRIPTION
**Pull request by tivisse**
Task : https://www.odoo.com/web#id=11363&view_type=form&model=project.task&action=333&active_id=248&menu_id=4720
PAD : https://pad.odoo.com/p/r.EDVLZWwA6kGFlmsg

### CODE REVIEW

File: addons/account/views/res_config_view.xml [hunk 1]
```diff
@@ -243,6 +243,6 @@
         </record>
 
         <menuitem id="menu_account_config" name="Settings" parent="menu_finance_configuration"
-            sequence="0" action="action_account_config" groups="base.group_configuration"/>
+            sequence="0" action="action_account_config" groups="base.group_no_one"/>
```
Comment (tde-banana-odoo) [inline addons/account/views/res_config_view.xml]: Intended ?

File: addons/account_accountant/account_accountant_data.xml [hunk 2]
```diff
@@ -8,7 +8,7 @@
             web_icon="account,static/description/icon.png"
             id="account.menu_finance"
             sequence="40"/>
-        <menuitem id="account.menu_account_config" name="Settings" parent="account.menu_finance_configuration" sequence="0" groups="base.group_configuration"/>
+        <menuitem id="account.menu_account_config" name="Settings" parent="account.menu_finance_configuration" sequence="0"/>
```
Comment (tde-banana-odoo) [inline addons/account_accountant/account_accountant_data.xml]: Are you based on the group no one of the parent ? Or should it be displayed everytime ?

File: addons/base_setup/res_config.py [hunk 3]
```diff
@@ -28,6 +28,10 @@ class base_config_settings(osv.osv_memory):
             help="Share your partners to all companies defined in your instance.\n"
                  " * Checked : Partners are visible for every companies, even if a company is defined on the partner.\n"
                  " * Unchecked : Each company can see only its partner (partners where company is defined). Partners not related to a company are visible for all companies."),
+        'group_multi_currency': fields.boolean('Allow multi currencies',
+            implied_group='base.group_multi_currency',
+            help="Allows you multi currency environment"),
```
Comment (tde-banana-odoo) [inline addons/base_setup/res_config.py]: I bet the "you" is not required. And I don't mean that you are not required. You know I love you.

File: addons/website_slides/view/website_slides.xml [hunk 4]
```diff
@@ -201,7 +201,7 @@
                     <t t-set="header_object" t-value="category"/>
                 </t>
             </t>
-            <section t-if="not channel.promoted_slide_id and not tag and not slide_type and not search and not is_public_user and not category and display_channel_settings" class="wrap mt16 mb0" groups="base.group_configuration">
+            <section t-if="not channel.promoted_slide_id and not tag and not slide_type and not search and not is_public_user and not category and display_channel_settings" class="wrap mt16 mb0">
```
Comment (tde-banana-odoo) [inline addons/website_slides/view/website_slides.xml]: No more groups ?

File: openerp/addons/base/res/res_users.py [hunk 5]
```diff
@@ -769,12 +779,27 @@ class UsersView(models.Model):
     @api.model
     def create(self, values):
         values = self._remove_reified_groups(values)
-        return super(UsersView, self).create(values)
+        user = super(UsersView, self).create(values)
+        group_multi_company = self.env['ir.model.data'].xmlid_to_object('base.group_multi_company')
+        if group_multi_company and 'company_ids' in values:
+            if len(user.company_ids) <= 1 and user.id in group_multi_company.users.ids:
+                group_multi_company.write({'users': [(3, user.id, _)]})
```
Comment (tde-banana-odoo) [inline openerp/addons/base/res/res_users.py]: (3, id) - (4, id)

File: addons/account/views/res_config_view.xml (hunk 1 above)
Comment (tivisse) [inline addons/account/views/res_config_view.xml]: Yep. A account manager shouldn't be able neither to install a new app/module nor to change the basic account configuration without the administration rights. This behavior has been applied on all App --> Configuration --> Settings menuitems.

Comment (tivisse) [inline addons/account/views/res_config_view.xml]: Forget what I said. I will check that everything's okay.

File: addons/website_slides/view/website_slides.xml (hunk 4 above)
Comment (tivisse) [inline addons/website_slides/view/website_slides.xml]: Maybe website designer, it's the only one that klopt according to me.

File: addons/mass_mailing/views/res_config.xml [hunk 6]
```diff
@@ -27,8 +27,8 @@
         </record>
 
         <menuitem id="menu_mass_mailing_configuration" name="Configuration" parent="mass_mailing_menu_root"
-            sequence="100" groups="base.group_configuration"/>
+            sequence="100"/>
         <menuitem id="menu_mass_mailing_global_settings" name="Settings" 
-            parent="menu_mass_mailing_configuration" sequence="0" action="action_mass_mailing_configuration"/>
+            parent="menu_mass_mailing_configuration" sequence="0" action="action_mass_mailing_configuration" groups="base.group_no_one"/>
```
Comment (tde-banana-odoo) [inline addons/mass_mailing/views/res_config.xml]: Not group system ?

File: addons/mrp/res_config_view.xml [hunk 7]
```diff
@@ -31,6 +31,6 @@
         </record>
 
         <menuitem id="base.menu_mrp_config" name="Settings" parent="menu_mrp_configuration"
-            sequence="0" action="action_mrp_configuration" groups="base.group_configuration"/>
+            sequence="0" action="action_mrp_configuration" groups="base.group_no_one"/>
```
Comment (tde-banana-odoo) [inline addons/mrp/res_config_view.xml]: Not group system ?

File: addons/project/res_config_view.xml [hunk 8]
```diff
@@ -47,6 +47,6 @@
         </record>
 
         <menuitem id="base.menu_project_general_settings" name="Settings" parent="menu_project_config"
-            sequence="0" action="action_config_settings"/>
+            sequence="0" action="action_config_settings" groups="base.group_no_one"/>
```
Comment (tde-banana-odoo) [inline addons/project/res_config_view.xml]: Not group system ?

File: addons/sale/res_config_view.xml [hunk 9]
```diff
@@ -49,9 +49,9 @@
         </record>
 
         <menuitem id="base.menu_sale_config" name="Configuration" parent="base.menu_base_partner"
-            sequence="100" groups="base.group_configuration"/>
+            sequence="100"/>
         <menuitem id="base.menu_sale_general_settings" name="Settings" parent="base.menu_sale_config"
-            sequence="0" action="base_setup.action_sale_config"/>
+            sequence="0" action="base_setup.action_sale_config" groups="base.group_no_one"/>
```
Comment (tde-banana-odoo) [inline addons/sale/res_config_view.xml]: Not group system ?

File: addons/utm/views/utm.xml [hunk 10]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
         <menuitem id="menu_link_tracker_root" name="Link Tracker" groups="base.group_no_one" sequence="30"/>
-        <menuitem name="UTMs" id="marketing_utm" groups="base.group_configuration" parent="menu_link_tracker_root" sequence="99"/>
+        <menuitem name="UTMs" id="marketing_utm" parent="menu_link_tracker_root" sequence="99"/>
```
Comment (tde-banana-odoo) [inline addons/utm/views/utm.xml]: Link Tracker accessible to everyone ?

File: openerp/addons/base/res/res_partner_view.xml [hunk 11]
```diff
@@ -8,9 +8,8 @@
             groups="base.group_sale_salesman"
             sequence="6"/>
 
-            <!-- This menu will disappear, I commented it for tests purpose -->
```
Comment (tde-banana-odoo) [inline openerp/addons/base/res/res_partner_view.xml]: Why removing this funny comment ? It is part of our valuable history !

File: openerp/addons/base/res/res_users.py [hunk 12]
```diff
@@ -201,7 +204,8 @@ def _default_groups(self):
     login_date = fields.Datetime(related='log_ids.create_date', string='Latest connection')
     share = fields.Boolean(compute='_compute_share', string='Share User', store=True,
          help="External user with limited access, created only for the purpose of sharing data.")
-
+    companies_count = fields.Integer(compute='_compute_companies_count', string="Number of Companies", default=_companies_count, store=False)
```
Comment (tde-banana-odoo) [inline openerp/addons/base/res/res_users.py]: store=False ?

File: openerp/addons/base/res/res_users.py [hunk 13]
```diff
@@ -769,12 +780,27 @@ class UsersView(models.Model):
     @api.model
     def create(self, values):
         values = self._remove_reified_groups(values)
-        return super(UsersView, self).create(values)
+        user = super(UsersView, self).create(values)
+        group_multi_company = self.env['ir.model.data'].xmlid_to_object('base.group_multi_company')
```
Comment (tde-banana-odoo) [inline openerp/addons/base/res/res_users.py]: ref !

File: openerp/addons/base/res/res_users.py [hunk 14]
```diff
@@ -769,12 +780,27 @@ class UsersView(models.Model):
...
+            elif len(user.company_ids) > 1 and user.id not in group_multi_company.users.ids:
+                group_multi_company.write({'users': [(4, user.id)]})
+        return user
 
     @api.multi
     def write(self, values):
         values = self._remove_reified_groups(values)
-        return super(UsersView, self).write(values)
+        res = super(UsersView, self).write(values)
+        group_multi_company = self.env['ir.model.data'].xmlid_to_object('base.group_multi_company')
```
Comment (tde-banana-odoo) [inline openerp/addons/base/res/res_users.py]: ref !

**Pull request merged**